### PR TITLE
DUP-360: Hold Pass Timer

### PR DESCRIPTION
**Pull request by danieltruong**
### GH Ticket:

BRS-360

### GH Ticket URL:

https://github.com/bcgov/parks-reso-api/issues/360

### Description:

**This is a breaking change that requires the API ticket DUP-336.**

This update adds the concept of a hold pass timer. Some notable changes include:

- Addition of Cloudflare Turnstile
- Addition of timer
- Removal of captcha on the contact page.

Note that in all environments, we need a new variable called TURNSTILE_SITE_KEY. This variable is the site key provided by Cloudflare.

### CODE REVIEW

File: src/app/services/pass.service.ts [hunk 1]
```diff
@@ -163,5 +164,33 @@ export class PassService {
...
+    } catch (err) {
+      this.loggerService.debug(`${err}`);
+      throw err;
+    }
+  }
+
+  getDecodedToken(token: string): any {
+    try {
+      return jwtDecode(token);
+    } catch (Error) {
```
Comment (marklise) [inline src/app/services/pass.service.ts:192]: This should be lowercase `error` not the `Error` class

Comment (danieltruong) [reply src/app/services/pass.service.ts]: [a525da5](https://github.com/bcgov/parks-reso-public/pull/379/commits/a525da548aab47695ecafbdc621346deb5f26e1b)

File: src/app/registration/facility-select/facility-select.component.ts [hunk 2]
```diff
@@ -26,6 +29,15 @@ export class FacilitySelectComponent implements OnInit {
...
   public showAsUnbookable = false;
+  public loading = false;
+
+  public token = null;
+  public renderTurnstile = false;
+  public siteKey = this.configService.config['TURNSTILE_SITE_KEY'];
+  // Always block
+  // public siteKey = "2x00000000000000000000AB"
+  // Force challenge
+  // public siteKey = "3x00000000000000000000FF"
```
Comment (cameronpettit) [inline src/app/registration/facility-select/facility-select.component.ts:40]: Do we want to leave these in here?

File: src/env.js [hunk 3]
```diff
@@ -15,6 +15,8 @@
   window.__env.API_PUBLIC_PATH = '/api';
   window.__env.ASSETS_S3_URL = 'https://d3ptawfth8sh71.cloudfront.net';
 
+  window.__env.TURNSTILE_SITE_KEY = '2x00000000000000000000AB';
+
```
Comment (cameronpettit) [inline src/env.js:19]: Commit this or make injectable secret?

Comment (cameronpettit) [reply src/env.js:19]: just a test token

**Review by cameronpettit — APPROVED** [review]

**Pull request merged**